File: client/app/src/main/java/ca/josephroque/hivemind/game/Movement.kt
package ca.josephroque.hivemind.game

/**
 * Copyright (C) 2019 Joseph Roque
 *
 * Describes a movement from one position to another.
 */
sealed class Movement {
    data class Move(val unit: Unit, val target: Position) : Movement()
    data class Yoink(val unit: Unit, val target: Position) : Movement()
    data class Place(val unit: Unit, val target: Position) : Movement()
}

File: engine/Sources/HiveMindEngine/State/Movement.swift
//
//  Movement.swift
//  HiveMindEngine
//
//  Created by Joseph Roque on 2019-02-07.
//  Copyright © 2019 Joseph Roque. All rights reserved.
//

import Foundation

public enum Movement: Hashable, Equatable {
	case move(unit: Unit, to: Position)
	case yoink(pillBug: Unit, unit: Unit, to: Position)
	case place(unit: Unit, at: Position)
}
